sbc_compassion/models/correspondence_template.py

```diff
@@ -181,6 +185,7 @@ def generate_pdf(self, pdf_name, header, text, image_data, background_list=None)
                 text_list.append(text_box.get_json_repr())
             overflow_template = [add_background.name, header_data, text_list, []]
 
+        callbacks.get("apply_text_callback", lambda: None)()
```

Comment:
Why is there a need to perform the callback like this only to update the progression? This introduces a lot of complexity in the code, but we could simply update the progession directly in the code or do you see some strong issues in doing like that?
```suggestion
        if s2b_generator is not None:
            s2b_generator.generation_status = "apply_text"
```
The generator would be passed in the arguments instead of the callbacks (see my other suggestion).